src/compiler.js

```diff
@@ -23,7 +23,9 @@ module.exports.run = ({ prefix, favicons: options, logo, cache: cacheDirectory }
 
   new SingleEntryPlugin(context, `!${cache}${loader}!${logo}`, path.basename(logo)).apply(compiler);
 
-  compilation.hooks.webappWebpackPluginBeforeEmit = new AsyncSeriesWaterfallHook(['result']);
+  if(compilation.hooks) {
+    compilation.hooks.webappWebpackPluginBeforeEmit = new AsyncSeriesWaterfallHook(['result']);
```

Comment:
We need to make sure we only `require('tappable')` when running on Webpack v4, so could you move `const AsyncSeriesWaterfallHook = require('tapable').AsyncSeriesWaterfallHook;` inside this conditional?